kubectl-plugin/pkg/cmd/version/version_test.go

```diff
@@ -0,0 +1,189 @@
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+
+	rayclient "github.com/ray-project/kuberay/ray-operator/pkg/client/clientset/versioned"
+)
+
+func TestRayVersionCheckContext(t *testing.T) {
+	testStreams, _, _, _ := genericclioptions.NewTestIOStreams()
+	testContext := "test-context"
+
+	kubeConfigWithCurrentContext, err := createTempKubeConfigFile(t, "my-fake-context")
+	assert.Nil(t, err)
+
+	kubeConfigWithoutCurrentContext, err := createTempKubeConfigFile(t, "")
+	assert.Nil(t, err)
+
+	tests := []struct {
+		name        string
+		opts        *VersionOptions
+		expectError string
+	}{
+		{
+			name: "Test validation when no context is set",
+			opts: &VersionOptions{
+				configFlags: genericclioptions.NewConfigFlags(false),
+				ioStreams:   &testStreams,
+			},
+			expectError: "no context is currently set, use \"--context\" or \"kubectl config use-context <context>\" to select a new one",
+		},
+		{
+			name: "no error when kubeconfig has current context and --context switch isn't set",
+			opts: &VersionOptions{
+				configFlags: &genericclioptions.ConfigFlags{
+					KubeConfig: &kubeConfigWithCurrentContext,
+				},
+				ioStreams: &testStreams,
+			},
+		},
+		{
+			name: "no error when kubeconfig has no current context and --context switch is set",
+			opts: &VersionOptions{
+				configFlags: &genericclioptions.ConfigFlags{
+					KubeConfig: &kubeConfigWithoutCurrentContext,
+					Context:    &testContext,
+				},
+				ioStreams: &testStreams,
+			},
+		},
+		{
+			name: "no error when kubeconfig has current context and --context switch is set",
+			opts: &VersionOptions{
+				configFlags: &genericclioptions.ConfigFlags{
+					KubeConfig: &kubeConfigWithCurrentContext,
+					Context:    &testContext,
+				},
+				ioStreams: &testStreams,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.opts.checkContext()
+			fmt.Printf("err: %v\n", err)
+			if tc.expectError != "" {
+				assert.Equal(t, tc.expectError, err.Error())
+			} else {
+				assert.Nil(t, err)
+			}
+		})
+	}
+}
+
+// createTempKubeConfigFile creates a temporary kubeconfig file with the given current context.
+func createTempKubeConfigFile(t *testing.T, currentContext string) (string, error) {
+	tmpDir := t.TempDir()
+
+	// Set up fake config for kubeconfig
+	config := &api.Config{
+		Clusters: map[string]*api.Cluster{
+			"test-cluster": {
+				Server:                "https://fake-kubernetes-cluster.example.com",
+				InsecureSkipTLSVerify: true, // For testing purposes
+			},
+		},
+		Contexts: map[string]*api.Context{
+			"my-fake-context": {
+				Cluster:  "my-fake-cluster",
+				AuthInfo: "my-fake-user",
+			},
+		},
+		CurrentContext: currentContext,
+		AuthInfos: map[string]*api.AuthInfo{
+			"my-fake-user": {
+				Token: "", // Empty for testing without authentication
+			},
+		},
+	}
+
+	fakeFile := filepath.Join(tmpDir, ".kubeconfig")
+
+	return fakeFile, clientcmd.WriteToFile(*config, fakeFile)
+}
```

Comment:
Again, let's wait for #2891 being merged and use the function in `kubectl-plugin/pkg/util/kubeconfig.go` directly.